OADP-3399: Reworking OpenShift KubeVirt OADP docs

anarnold97 opened the pull request:
### JIRA 

* [OADP-3399](https://issues.redhat.com/browse/OADP-3399)

    * Adding KubeVirt to DPA and including Virt docs into OADP docs.

    * Fixed the broken ifdef statement on 1.3 DPA for ODF 

* To do :

    * Check the other parts of the Virt docs for what we should bring over to OADP

GA - 16th April

### Versions

* OCP 4.13 → branch/enterprise-4.13
* OCP 4.14 → branch/enterprise-4.14
* OCP 4.15 → branch/enterprise-4.15
*  OCP 4.16 → branch/enterprise-4.16


### Link to docs preview:

* [Configuring the OpenShift API for Data Protection with OpenShift Virtualization](https://72372--ocpdocs-pr.netlify.app/openshift-enterprise/latest/backup_and_restore/application_backup_and_restore/installing/installing-oadp-kubevirt)

### QE review:
- [ ] QE has approved this change.
<!--- QE approval is required to merge a PR except for changes that do not impact the meaning of the docs. --->

Additional information:
<!--- Optional: Include additional context or expand the description here.--->

<!--- After you open your PR, ask for review from the OpenShift docs team:
  For community authors: Tag @openshift/team-documentation in a GitHub comment.--->

anarnold97 commented:
/rest

anarnold97 commented:
/label oadp

anarnold97 commented on backup_and_restore/application_backup_and_restore/installing/installing-oadp-kubevirt.adoc:
```diff
@@ -0,0 +1,68 @@
+:_mod-docs-content-type: ASSEMBLY
+[id="installing-oadp-kubevirt"]
+// If docs team approves the name, "Configuring OADP with Kubevirt" great but I suspect it will need to be "Configuring OADP with OpenShift Virtualization" or "Configuring OADP with Virtualization" or perhaps... bear with me.. "OADP and OpenShift-Virtualization"
+= Configuring the OpenShift API for Data Protection with {VirtProductName}
```
We would like to call it, but think reviewer will have an objection
 
```suggestion
= Configuring the OpenShift API for Data Protection with KubeVirt
```

weshayutin replied:
@anarnold97 did we get any feedback on this?  IMHO my preference from first pref to last, would be.
1. OpenShift Virtualization
2. Virtualization
3. KubeVirt

anarnold97 replied:
Not pushed it to OCP review yet. As soon as you and Amos give an LGTM, I will push it

Cool, let us go with Configuring the OpenShift API for Data Protection with {VirtProductName}

{VirtProductName} -> OpenShift Virtualization

anarnold97 commented:
/label oadp

anarnold97 commented on modules/oadp-installing-dpa-1-3.adoc:
```diff
@@ -142,18 +147,19 @@ spec:
+<6> Set this value to `true` if you want to enable `nodeAgent` and perform File System Backup.
+<7> Enter `kopia` or `restic` as your uploader. You cannot change the selection after the installation. For the Built-in DataMover you must use Kopia. The `nodeAgent` deploys a daemon set, which means that the `nodeAgent` pods run on each working node. You can configure File System Backup by adding `spec.defaultVolumesToFsBackup: true` to the `Backup` CR.
+<8> Specify the nodes on which Kopia or Restic are available. By default, Kopia or Restic run on all nodes.
+<9> Specify the Azure resource group.
```
```suggestion
<9> Specify the Microsoft Azure resource group.
```

anarnold97 commented on modules/oadp-installing-dpa-1-3.adoc:
```diff
@@ -142,18 +147,19 @@ spec:
+<7> Enter `kopia` or `restic` as your uploader. You cannot change the selection after the installation. For the Built-in DataMover you must use Kopia. The `nodeAgent` deploys a daemon set, which means that the `nodeAgent` pods run on each working node. You can configure File System Backup by adding `spec.defaultVolumesToFsBackup: true` to the `Backup` CR.
+<8> Specify the nodes on which Kopia or Restic are available. By default, Kopia or Restic run on all nodes.
+<9> Specify the Microsoft Azure resource group.
+<10> Specify the Azure storage account ID.
```
```suggestion
<10> Specify the Microsoft Azure storage account ID.
```

anarnold97 commented on modules/oadp-installing-dpa-1-3.adoc:
```diff
@@ -142,18 +147,19 @@ spec:
+<8> Specify the nodes on which Kopia or Restic are available. By default, Kopia or Restic run on all nodes.
+<9> Specify the Microsoft Azure resource group.
+<10> Specify the Microsoft Azure storage account ID.
+<11> Specify the Azure subscription ID.
```
```suggestion
<11> Specify the Microsoft Azure subscription ID.
```

anarnold97 commented on _topic_maps/_topic_map.yml:
```diff
@@ -3056,6 +3056,8 @@ Topics:
       File: installing-oadp-mcg
     - Name: Configuring OADP with ODF
       File: installing-oadp-ocs
+    - Name: Configuring OADP with KubeVirt
```
```suggestion
    - Name: Configuring OADP with OpenShift Virtualization
```

anarnold97 commented:
/test validate-asciidoc

anarnold97 commented on snippets/oadp-virtualization-supported-backup-types.adoc:
```diff
@@ -0,0 +1,21 @@
+
+For more information, see xref:../../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/oadp-backing-up-applications-restic-doc.adoc#oadp-backing-up-applications-restic-doc[Backing up applications with File System Backup: Kopia or Restic].
+====
+
```
```suggestion
```

anarnold97 commented on snippets/oadp-virtualization-supported-backup-types.adoc:
```diff
@@ -0,0 +1,20 @@
+====
+(oadp-full} with {VirtProductName} backup and restore support is limited to:
+
+* Container Storage Interface (CSI) backups
```
```suggestion
* Container Storage Interface (CSI) backups

```

anarnold97 commented on snippets/oadp-virtualization-supported-backup-types.adoc:
```diff
@@ -0,0 +1,21 @@
+This support excludes:
+
+* File System backup and restore.
+* Volume Snapshot Backups and Restores.
```
```suggestion

* Volume Snapshot Backups and Restores.
```

weshayutin reviewed (approved):
/LGTM

amastbau commented on backup_and_restore/application_backup_and_restore/installing/installing-oadp-kubevirt.adoc:
```diff
@@ -0,0 +1,77 @@
+
+toc::[]
+
+You install the {oadp-first} with {VirtProductName} by installing the OADP Operator and configuring a backup location and a snapshot location. Then, you install the Data Protection Application.
```
I think we can remove any reference to VSL since we only support CSI + CSI w/ DM

anarnold97 replied:
```suggestion
You install the {oadp-first} with {VirtProductName} by installing the OADP Operator and configuring a backup location. Then, you install the Data Protection Application.
```

amastbau commented on modules/oadp-kubevirt-setting-resource-limits-and-requests.adoc:
```diff
@@ -0,0 +1,48 @@
+
+[NOTE]
+====
+Kopia is an option in OADP 1.3 and later releases. You can use Kopia for file system backups, and Kopia is your only option for Data Mover cases with the built-in Data Mover.
```
i think we need to remove refferences to filesystem as well

anarnold97 replied:
```suggestion
Kopia is an option in OADP 1.3 and later releases. Kopia is your only option for Data Mover cases with the built-in Data Mover.
```

amastbau commented on modules/oadp-kubevirt-setting-resource-limits-and-requests.adoc:
```diff
@@ -0,0 +1,48 @@
+        nodeSelector: <node selector> <1>
+        resourceAllocations: <2>
+          limits:
+            cpu: "1"
+            memory: 1024Mi
+          requests:
+            cpu: 200m
+            memory: 256Mi
```
Are those coming from scale teams?

anarnold97 replied:
So these values are coming from the Virt docs - https://docs.openshift.com/container-platform/4.16//virt/backup_restore/virt-installing-configuring-oadp.html

anarnold97 replied:
Removed

amastbau commented on backup_and_restore/application_backup_and_restore/installing/installing-oadp-kubevirt.adoc:
```diff
@@ -0,0 +1,77 @@
+:context: installing-oadp-kubevirt
+:installing-oadp-kubevirt:
+:credentials: cloud-credentials
+:provider: kubevirt
```
why is the provider name, kubevirt? :)

anarnold97 replied:
Do i need to change it? Nobody sees it apart from us?

skrthomas replied:
If kubivert is technically not accurate, maybe it shouldn't be here? Not an expert here, but noticed this was a resolved comment that wasn't outdated and thought I'd re-raise it.

skrthomas replied:
If these files are inherited by someone later, incorrect metadata could lead to confusion

anarnold97 commented:
@amastbau - have made your suggested changes.

In [modules/oadp-kubevirt-setting-resource-limits-and-requests.adoc](https://github.com/openshift/openshift-docs/pull/72372/files#diff-487d39e2e7968518789a096a121f658801d03fa738dd52b46b735c24b09ff133) the values are coming from the [Virt docs](https://docs.openshift.com/container-platform/4.16//virt/backup_restore/virt-installing-configuring-oadp.html)
I will speak to Tzahi and Mordecchai and ask for their opinion, but might need to make the changes post-GA

anarnold97 commented on backup_and_restore/application_backup_and_restore/installing/installing-oadp-kubevirt.adoc:
```diff
@@ -0,0 +1,77 @@
+//include::modules/oadp-installing-operator.adoc[leveloffset=+1]
+// include::modules/oadp-about-backup-snapshot-locations-secrets.adoc[leveloffset=+1]
+
+include::modules/oadp-creating-default-secret.adoc[leveloffset=+2]
```
```suggestion
// include::modules/oadp-creating-default-secret.adoc[leveloffset=+2]
```

anarnold97 commented on backup_and_restore/application_backup_and_restore/installing/installing-oadp-kubevirt.adoc:
```diff
@@ -0,0 +1,77 @@
+
+// include::modules/oadp-creating-default-secret.adoc[leveloffset=+2]
+
+include::modules/oadp-secrets-for-different-credentials.adoc[leveloffset=+2]
```
```suggestion
// include::modules/oadp-secrets-for-different-credentials.adoc[leveloffset=+2]
```

weshayutin reviewed (approved):
/LGTM

amastbau commented:
/LGTM

anarnold97 commented:
@amastbau - thanks very much. I won't look for you

anarnold97 commented:
/label peer-review-needed

skrthomas commented on snippets/oadp-virtualization-supported-backup-types.adoc:
```diff
@@ -0,0 +1,22 @@
+
+* Volume Snapshot Backups and Restores.
+
+For more information, see xref:../../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/oadp-backing-up-applications-restic-doc.adoc#oadp-backing-up-applications-restic-doc[Backing up applications with File System Backup: Kopia or Restic].
```
text snippets cannot contain xrefs:
https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#writing-text-snippets

I would suggest making this an .Additional resources section: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#lead-in-link-sentences

skrthomas commented on snippets/oadp-virtualization-supported-backup-types.adoc:
```diff
@@ -0,0 +1,22 @@
+// Text snippet included in the following assemblies:
+//
+// ../backup_and_restore/application_backup_and_restore/installing/installing-oadp-kubevirt.adoc
```
```suggestion
// * <full-file-path>/backup_and_restore/application_backup_and_restore/installing/installing-oadp-kubevirt.adoc
```

skrthomas replied:
Instead of the `..` you should include the full filepath, and add a * per the https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#text-snippet-file-metadata

anarnold97 replied:
Except adding snippets in OCP docs does not work consistently. 

So we are probably going to stop sing snippets as there is no guarantee they will work

skrthomas commented on modules/oadp-kubevirt-setting-resource-limits-and-requests.adoc:
```diff
@@ -0,0 +1,48 @@
+  configuration:
+    velero:
+      podConfig:
+        nodeSelector: <node selector> <1>
```
```suggestion
        nodeSelector: <node_selector> <1>
```

skrthomas replied:
If this is a user-replaced value, use the following syntax: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#command-syntax-for-replaceable-values

skrthomas commented on modules/oadp-kubevirt-setting-resource-limits-and-requests.adoc:
```diff
@@ -0,0 +1,48 @@
+// * backup_and_restore/application_backup_and_restore/installing/installing-oadp-kubevirt.adoc
+
+
+:_mod-docs-content-type: CONCEPT
```
```suggestion
:_mod-docs-content-type: PROCEDURE
```

skrthomas replied:
I think this seems like a PROCEDURE? Maybe mis-categorized?

skrthomas commented on backup_and_restore/application_backup_and_restore/installing/installing-oadp-mcg.adoc:
```diff
@@ -41,4 +41,11 @@ include::modules/oadp-installing-dpa-1-2-and-earlier.adoc[leveloffset=+1]
 include::modules/oadp-installing-dpa-1-3.adoc[leveloffset=+1]
 include::modules/oadp-enabling-csi-dpa.adoc[leveloffset=+2]
 
+.Additional resources
```
This is a global comment because I see this in a few places, but the .Additional resources sections should only contain text inside the xref:...[only text here]. In Customer Portal, the other text doesn't render properly. See this for more info: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#additional-resources-sections

skrthomas replied:
Also, the "see" construction for a list doesn't follow IBMSG recommendations for lead-in wording: https://www.ibm.com/docs/en/ibm-style?topic=format-lists#lead-in-wording

skrthomas commented on backup_and_restore/application_backup_and_restore/installing/installing-oadp-kubevirt.adoc:
```diff
@@ -0,0 +1,77 @@
+
+toc::[]
+
+You install the {oadp-first} with {VirtProductName} by installing the OADP Operator and configuring a backup location. Then, you install the Data Protection Application.
```
```suggestion
You can install the {oadp-first} with {VirtProductName} by installing the OADP Operator and configuring a backup location. Then, you can install the Data Protection Application.
```

skrthomas reviewed (commented):
A few mod docs and OCP contributing guideline things to fix here. Looking good!

anarnold97 commented on backup_and_restore/application_backup_and_restore/installing/installing-oadp-aws.adoc:
```diff
@@ -35,4 +35,11 @@ include::modules/oadp-installing-dpa-1-2-and-earlier.adoc[leveloffset=+1]
 
+.Additional resources
+
+For more information on installing OADP to back up and restore virtual machines, see:
+
```
```suggestion
```

anarnold97 commented on backup_and_restore/application_backup_and_restore/installing/installing-oadp-azure.adoc:
```diff
@@ -35,4 +35,10 @@ include::modules/oadp-installing-dpa-1-2-and-earlier.adoc[leveloffset=+1]
 
+.Additional resources
+
+For more information on installing OADP to back up and restore virtual machines, see:
+
```
```suggestion
```

anarnold97 commented on backup_and_restore/application_backup_and_restore/installing/installing-oadp-gcp.adoc:
```diff
@@ -36,4 +36,11 @@ include::modules/oadp-gcp-wif-cloud-authentication.adoc[leveloffset=+1]
 
+.Additional resources
+
+For more information on installing OADP to back up and restore virtual machines, see:
+
```
```suggestion
```

anarnold97 commented on backup_and_restore/application_backup_and_restore/installing/installing-oadp-kubevirt.adoc:
```diff
@@ -0,0 +1,77 @@
+
+* Access to the cluster as a user with the `cluster-admin` role.
+
+.Procedure
+
+. Install the xref:../../../backup_and_restore/application_backup_and_restore/installing/about-installing-oadp.adoc#about-installing-oadp[OADP Operator] according to the instructions for your storage provider.
+
+. Install the xref:oadp-installing-dpa-1-3_installing-oadp-kubevirt[Data Protection Application] with the `kubevirt` and `openshift` xref:../../../backup_and_restore/application_backup_and_restore/oadp-features-plugins.adoc#oadp-plugins_oadp-features-plugins[plugins].
+
+. Back up virtual machines by creating a xref:../../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/backing-up-applications.adoc#backing-up-applications[`Backup` custom resource (CR)].
```
There will probably be sections added post-GA

anarnold97 replied:
Please can I come back next week after GA and fix this

skrthomas replied:
I checked on this with the rest of the peer review squad, and the squad agrees merging this is not okay because it does not fit with mod docs standards for assemblies: https://redhat-documentation.github.io/modular-docs/#assembly-guidelines

Assemblies can have introductions, prerequisites, and modules as the main pieces. If you have a procedure, it belongs in a procedure module. I'm not sure what other post-GA sections would validate having a procedure in an assembly 🤷‍♀️ because they just aren't allowed.

anarnold97 commented:
/label merge-review-needed

anarnold97 commented on modules/oadp-kubevirt-setting-resource-limits-and-requests.adoc:
```diff
@@ -0,0 +1,48 @@
+            memory: 1024Mi
+          requests:
+            cpu: 200m
+            memory: 256Mi
```
Removed from PR

anarnold97 commented:
/remove merge-review-needed

anarnold97 commented:
/remove-label merge-review-needed

anarnold97 commented on backup_and_restore/application_backup_and_restore/installing/installing-oadp-kubevirt.adoc:
```diff
@@ -0,0 +1,75 @@
+
+// include::modules/oadp-kubevirt-setting-resource-limits-and-requests.adoc[leveloffset=+2]
+
+// include::modules/oadp-self-signed-certificate.adoc[leveloffset=+2]
```
```suggestion
```

sheriff-rh commented:
@anarnold97 This doesn't quite look ready for merge. 
1. Please squash the commits.
2. Which versions does this apply to?

anarnold97 commented:
Apologies, updated versions and squashed

anarnold97 commented:
/label merge-review-needed

I have another PR I will be working on post-GA and can include updates to that. See [OADP-3617](https://issues.redhat.com/browse/OADP-3617)

skrthomas commented:
@anarnold97 I'm going to hold merge due to this [comment](https://github.com/openshift/openshift-docs/pull/72372#discussion_r1567809918) being unresolved. This is a modular docs requirement, and this PR can't be merged until the comment is resolved.

skrthomas commented on backup_and_restore/application_backup_and_restore/installing/installing-oadp-kubevirt.adoc:
```diff
@@ -0,0 +1,71 @@
+
+* Access to the cluster as a user with the `cluster-admin` role.
+
+.Procedure
+
+. Install the xref:../../../backup_and_restore/application_backup_and_restore/installing/about-installing-oadp.adoc#about-installing-oadp[OADP Operator] according to the instructions for your storage provider.
+
+. Install the xref:oadp-installing-dpa-1-3_installing-oadp-kubevirt[Data Protection Application] with the `kubevirt` and `openshift` xref:../../../backup_and_restore/application_backup_and_restore/oadp-features-plugins.adoc#oadp-plugins_oadp-features-plugins[plugins].
+
+. Back up virtual machines by creating a xref:../../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/backing-up-applications.adoc#backing-up-applications[`Backup` custom resource (CR)].
+
++
+[WARNING]
+====
+Red Hat support is limited to only the following options:
+* CSI backups
+* CSI backups with DataMover.
+====
+
+. Restore the `Backup` CR by creating a xref:../../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/restoring-applications.adoc#restoring-applications[`Restore` CR].
+
+To install the OADP Operator in a restricted network environment, you must first disable the default OperatorHub sources and mirror the Operator catalog. For details, see xref:../../../operators/admin/olm-restricted-networks.adoc#olm-restricted-networks[Using Operator Lifecycle Manager on restricted networks].
```
```suggestion

```

skrthomas replied:
I see you're trying to do an Overview-type link situation here, but I think these kind of links need to be in == Additional resource heading like here: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#additional-resources-sections

maxwelldb commented on backup_and_restore/application_backup_and_restore/installing/installing-oadp-kubevirt.adoc:
```diff
@@ -0,0 +1,71 @@
+
+Back up and restore virtual machines by using the xref:../../../backup_and_restore/index.adoc#application-backup-restore-operations-overview[OpenShift API for Data Protection (OADP)].
+
+[NOTE]
+====
+(oadp-full} with {VirtProductName} backup and restore support is limited to:
+* Container Storage Interface (CSI) backups
+* Container Storage Interface (CSI) backups with DataMover
+
+This support excludes:
+* File System backup and restore.
+* Volume Snapshot Backups and Restores.
+
+For more information, see xref:../../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/oadp-backing-up-applications-restic-doc.adoc#oadp-backing-up-applications-restic-doc[Backing up applications with File System Backup: Kopia or Restic].
+====
```
The lists in this admonition do not render correctly in the preview. The attribute also needs to go between two brackets.

Currently:
![image](https://github.com/openshift/openshift-docs/assets/50179998/f28b054c-ffeb-48fb-9e2d-5aebe3c68b57)

This must be fixed prior to merging.

maxwelldb replied:
```suggestion
[NOTE]
====
{oadp-full} with {VirtProductName} backup and restore support is limited to:

* Container Storage Interface (CSI) backups
* Container Storage Interface (CSI) backups with DataMover

This support excludes:

* File System backup and restore.
* Volume Snapshot Backups and Restores.

For more information, see xref:../../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/oadp-backing-up-applications-restic-doc.adoc#oadp-backing-up-applications-restic-doc[Backing up applications with File System Backup: Kopia or Restic].
====
```

skrthomas reviewed (commented):
I added a suggestion about your .Procedure that's maybe more appropriate than pulling it into a procedure.

skrthomas commented on backup_and_restore/application_backup_and_restore/installing/installing-oadp-kubevirt.adoc:
```diff
@@ -0,0 +1,71 @@
+
+To install the OADP Operator in a restricted network environment, you must first disable the default OperatorHub sources and mirror the Operator catalog. For details, see xref:../../../operators/admin/olm-restricted-networks.adoc#olm-restricted-networks[Using Operator Lifecycle Manager on restricted networks].
+
+[id="configuring-dpa-kubevirt"]
+== Configuring the Data Protection Application
+
+// You can configure the Data Protection Application by setting Velero resource allocations or enabling self-signed CA certificates.
+
+include::modules/oadp-installing-dpa-1-3.adoc[leveloffset=+1]
```
I also notice that this is a bit contradictory? Its a `== Configuring`  heading but then an Installing module included? And in the preview, notice how the levels are the same? 
![image](https://github.com/openshift/openshift-docs/assets/77501157/1cb66202-12ac-4278-b1bf-ecd6bd272f1e)

skrthomas replied:
I'm not sure if this is installing or configuring, but I think this area needs to be re-worked. A leveloffset=+2 applied so that the headers are at different levels and also appropriate titling for the included content.

anarnold97 commented on backup_and_restore/application_backup_and_restore/installing/installing-oadp-kubevirt.adoc:
```diff
@@ -0,0 +1,82 @@
+
+To install the OADP Operator in a restricted network environment, you must first disable the default OperatorHub sources and mirror the Operator catalog. For details, see xref:../../../operators/admin/olm-restricted-networks.adoc#olm-restricted-networks[Using Operator Lifecycle Manager on restricted networks].
+
+[id="configuring-dpa-kubevirt"]
+== Configuring the Data Protection Application
```
```suggestion
```

anarnold97 commented on backup_and_restore/application_backup_and_restore/installing/installing-oadp-kubevirt.adoc:
```diff
@@ -0,0 +1,80 @@
+For more information, see xref:../../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/oadp-backing-up-applications-restic-doc.adoc#oadp-backing-up-applications-restic-doc[Backing up applications with File System Backup: Kopia or Restic].
+====
+
+[id="install-and configure-oadp-kubevirt"]
```
```suggestion
[id="install-and-configure-oadp-kubevirt_context"]
```

anarnold97 commented on backup_and_restore/application_backup_and_restore/installing/installing-oadp-kubevirt.adoc:
```diff
@@ -0,0 +1,80 @@
++
+[WARNING]
+====
+Red Hat support is limited to only the following options:
+* CSI backups
+* CSI backups with DataMover.
```
```suggestion
Red Hat support is limited to only the following options:

* CSI backups

* CSI backups with DataMover.
```

anarnold97 commented on backup_and_restore/application_backup_and_restore/installing/installing-oadp-kubevirt.adoc:
```diff
@@ -0,0 +1,82 @@
+[IMPORTANT]
+.OADP 1.2
+====
+Red Hat only supports the combination of OADP versions 1.3.0 and later, and {VirtProductName} versions 4.14 and later.
+OADP versions before 1.3.0 are not supported for back up and restore of {VirtProductName}.
```
```suggestion
Red Hat only supports the combination of {oadp-short} versions 1.3.0 and later, and {VirtProductName} versions 4.14 and later.

{oadp-short} versions before 1.3.0 are not supported for back up and restore of {VirtProductName}.
```

anarnold97 commented on backup_and_restore/application_backup_and_restore/installing/installing-oadp-kubevirt.adoc:
```diff
@@ -0,0 +1,82 @@
+
+. Install the xref:../../../backup_and_restore/application_backup_and_restore/installing/about-installing-oadp.adoc#about-installing-oadp[OADP Operator] according to the instructions for your storage provider.
+
+. Install the xref:oadp-installing-dpa-1-3_installing-oadp-kubevirt[Data Protection Application] with the `kubevirt` and `openshift` xref:../../../backup_and_restore/application_backup_and_restore/oadp-features-plugins.adoc#oadp-plugins_oadp-features-plugins[plugins].
```
```suggestion
. Install the xref:oadp-installing-dpa-1-3_installing-oadp-kubevirt[Data Protection Application (DPA)] with the `kubevirt` and `openshift` xref:../../../backup_and_restore/application_backup_and_restore/oadp-features-plugins.adoc#oadp-plugins_oadp-features-plugins[plugins].
```

anarnold97 commented on backup_and_restore/application_backup_and_restore/installing/installing-oadp-kubevirt.adoc:
```diff
@@ -0,0 +1,83 @@
+include::modules/oadp-installing-dpa-1-3.adoc[leveloffset=+1]
+
+[IMPORTANT]
+.OADP 1.2
```
```suggestion
.{oadp-short} 1.2
```

anarnold97 commented on backup_and_restore/application_backup_and_restore/installing/installing-oadp-kubevirt.adoc:
```diff
@@ -0,0 +1,83 @@
+:_mod-docs-content-type: ASSEMBLY
+[id="installing-oadp-kubevirt"]
+// If docs team approves the name, "Configuring OADP with Kubevirt" great but I suspect it will need to be "Configuring OADP with OpenShift Virtualization" or "Configuring OADP with Virtualization" or perhaps... bear with me.. "OADP and OpenShift-Virtualization"
+= Configuring the {oadp-full} API for Data Protection with {VirtProductName}
```
```suggestion
= Configuring the {oadp-full} with {VirtProductName}
```

anarnold97 commented on backup_and_restore/application_backup_and_restore/installing/installing-oadp-kubevirt.adoc:
```diff
@@ -0,0 +1,83 @@
+
+// include::snippets/oadp-mtc-operator.adoc[]
+
+Back up and restore virtual machines by using the xref:../../../backup_and_restore/index.adoc#application-backup-restore-operations-overview[OpenShift API for Data Protection (OADP)].
```
```suggestion
Back up and restore virtual machines by using the xref:../../../backup_and_restore/index.adoc#application-backup-restore-operations-overview[{oadp-full}].
```

anarnold97 commented on backup_and_restore/application_backup_and_restore/installing/installing-oadp-kubevirt.adoc:
```diff
@@ -0,0 +1,83 @@
+
+. Restore the `Backup` CR by creating a xref:../../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/restoring-applications.adoc#restoring-applications[`Restore` CR].
+
+To install the OADP Operator in a restricted network environment, you must first disable the default OperatorHub sources and mirror the Operator catalog. For details, see xref:../../../operators/admin/olm-restricted-networks.adoc#olm-restricted-networks[Using Operator Lifecycle Manager on restricted networks].
```
```suggestion
To install the {oadp-short} Operator in a restricted network environment, you must first disable the default OperatorHub sources and mirror the Operator catalog. For details, see xref:../../../operators/admin/olm-restricted-networks.adoc#olm-restricted-networks[Using Operator Lifecycle Manager on restricted networks].
```

anarnold97 commented on backup_and_restore/application_backup_and_restore/installing/installing-oadp-kubevirt.adoc:
```diff
@@ -0,0 +1,83 @@
+[id="install-and-configure-oadp-kubevirt_context"]
+== Installing and configuring {oadp-short} with {VirtProductName}
+
+As a cluster administrator, you install the {oadp-short} by installing the {oadp-short} Operator. 
```
```suggestion
As a cluster administrator, you install {oadp-short} by installing the {oadp-short} Operator. 
```

anarnold97 commented:
Corrected and updated to include a new level 2 header 

/label merge-review-needed

anarnold97 commented on backup_and_restore/application_backup_and_restore/installing/installing-oadp-kubevirt.adoc:
```diff
@@ -0,0 +1,83 @@
+
+As a cluster administrator, you install {oadp-short} by installing the {oadp-short} Operator. 
+
+The Operator installs link:https://velero.io/docs/v{velero-version}[Velero velero-version].
```
```suggestion
The Operator installs link:https://velero.io/docs/v{velero-version}[Velero {velero-version}].
```

ShaunaDiaz commented on backup_and_restore/application_backup_and_restore/installing/installing-oadp-mcg.adoc:
```diff
@@ -41,4 +41,11 @@ include::modules/oadp-installing-dpa-1-2-and-earlier.adoc[leveloffset=+1]
 
+.Additional resources
+
+For more information on installing OADP to back up and restore virtual machines, see:
```
Please fix all of the unaddressed comments on this PR before resubmitting for merge review.

ShaunaDiaz commented:
/remove-label merge-review-needed

skrthomas commented on modules/install-and-configure-oadp-kubevirt.adoc:
```diff
@@ -0,0 +1,23 @@
+
+.Procedure
+
+. Install the xref:../../../backup_and_restore/application_backup_and_restore/installing/about-installing-oadp.adoc#about-installing-oadp[OADP Operator] according to the instructions for your storage provider.
+
+. Install the xref:oadp-installing-dpa-1-3_installing-oadp-kubevirt[Data Protection Application (DPA)] with the `kubevirt` and `openshift` xref:../../../backup_and_restore/application_backup_and_restore/oadp-features-plugins.adoc#oadp-plugins_oadp-features-plugins[plugins].
+
+. Back up virtual machines by creating a xref:../../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/backing-up-applications.adoc#backing-up-applications[`Backup` custom resource (CR)].
```
You can't have xrefs in a module, which is why your build is broken. In my comment yesterday, I suggested that you consider pulling these links into an `==Additional resources` heading instead of categorizing them as a `.Procedure`, which they aren't. This entire "procedure" is more like an overview of procedures to come.

anarnold97 commented on backup_and_restore/application_backup_and_restore/installing/installing-oadp-mcg.adoc:
```diff
@@ -42,8 +42,12 @@ include::modules/oadp-installing-dpa-1-2-and-earlier.adoc[leveloffset=+1]
 .Additional resources
 
-* link:https://access.redhat.com/solutions/6719951[Performance tuning guide for Multicloud Object Gateway].
+* xref:../../../backup_and_restore/application_backup_and_restore/installing/installing-oadp-kubevirt.adoc#oadp-installing-dpa-1-3_installing-oadp-kubevirt[Installing the Data Protection Application] with the `kubevirt` and `openshift` xref:../../../backup_and_restore/application_backup_and_restore/oadp-features-plugins.adoc#oadp-plugins_oadp-features-plugins[plugins].
```
```suggestion
* link:https://access.redhat.com/solutions/6719951[Performance tuning guide for Multicloud Object Gateway]

* xref:../../../backup_and_restore/application_backup_and_restore/installing/installing-oadp-kubevirt.adoc#oadp-installing-dpa-1-3_installing-oadp-kubevirt[Installing the Data Protection Application] with the `kubevirt` and `openshift` xref:../../../backup_and_restore/application_backup_and_restore/oadp-features-plugins.adoc#oadp-plugins_oadp-features-plugins[plugins]
```

anarnold97 commented on backup_and_restore/application_backup_and_restore/installing/installing-oadp-kubevirt.adoc:
```diff
@@ -0,0 +1,61 @@
+. xref:../../../backup_and_restore/application_backup_and_restore/oadp-features-plugins.adoc#oadp-plugins_oadp-features-plugins[{oadp-short} plugins]
+. xref:../../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/backing-up-applications.adoc#backing-up-applications[Creating a `Backup` custom resource (CR)]
+. xref:../../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/restoring-applications.adoc#restoring-applications[Creating a `Restore` CR]
+.  xref::../../../operators/admin/olm-restricted-networks.adoc#olm-restricted-networks[Using Operator Lifecycle Manager on restricted networks]
```
```suggestion
.  xref::../../operators/admin/olm-restricted-networks.adoc#olm-restricted-networks[Using Operator Lifecycle Manager on restricted networks]
```

anarnold97 commented:
/remove-label peer-review-done

anarnold97 commented on backup_and_restore/application_backup_and_restore/installing/installing-oadp-aws.adoc:
```diff
@@ -38,4 +38,9 @@ include::modules/oadp-installing-dpa-1-2-and-earlier.adoc[leveloffset=+1]
 
+.Additional resources
+
+* xref:../../../backup_and_restore/application_backup_and_restore/installing/installing-oadp-kubevirt.adoc#oadp-installing-dpa-1-3_installing-oadp-kubevirt[Installing the Data Protection Application with the `kubevirt` and `openshift`]
```
```suggestion
* xref:../../../backup_and_restore/application_backup_and_restore/installing/installing-oadp-kubevirt.adoc#oadp-installing-dpa-1-3_installing-oadp-kubevirt[Installing the Data Protection Application with the `kubevirt` and `openshift` plugins]
```

anarnold97 commented on backup_and_restore/application_backup_and_restore/installing/installing-oadp-azure.adoc:
```diff
@@ -35,4 +35,8 @@ include::modules/oadp-installing-dpa-1-2-and-earlier.adoc[leveloffset=+1]
 
+.Additional resources
+
+* xref:../../../backup_and_restore/application_backup_and_restore/installing/installing-oadp-kubevirt.adoc#oadp-installing-dpa-1-3_installing-oadp-kubevirt[Installing the Data Protection Application with the `kubevirt` and `openshift`]
```
```suggestion
* xref:../../../backup_and_restore/application_backup_and_restore/installing/installing-oadp-kubevirt.adoc#oadp-installing-dpa-1-3_installing-oadp-kubevirt[Installing the Data Protection Application with the `kubevirt` and `openshift` plugins]
```

anarnold97 commented on backup_and_restore/application_backup_and_restore/installing/installing-oadp-gcp.adoc:
```diff
@@ -36,4 +36,9 @@ include::modules/oadp-gcp-wif-cloud-authentication.adoc[leveloffset=+1]
 
+.Additional resources
+
+* xref:../../../backup_and_restore/application_backup_and_restore/installing/installing-oadp-kubevirt.adoc#oadp-installing-dpa-1-3_installing-oadp-kubevirt[Installing the Data Protection Application with the `kubevirt` and `openshift`]
```
```suggestion
* xref:../../../backup_and_restore/application_backup_and_restore/installing/installing-oadp-kubevirt.adoc#oadp-installing-dpa-1-3_installing-oadp-kubevirt[Installing the Data Protection Application with the `kubevirt` and `openshift` plugins]
```

anarnold97 commented on backup_and_restore/application_backup_and_restore/installing/installing-oadp-mcg.adoc:
```diff
@@ -42,8 +42,14 @@ include::modules/oadp-installing-dpa-1-2-and-earlier.adoc[leveloffset=+1]
-* link:https://access.redhat.com/solutions/6719951[Performance tuning guide for Multicloud Object Gateway].
+* link:https://access.redhat.com/solutions/6719951[Performance tuning guide for Multicloud Object Gateway]
+
+* xref:../../../backup_and_restore/application_backup_and_restore/installing/installing-oadp-kubevirt.adoc#oadp-installing-dpa-1-3_installing-oadp-kubevirt[Installing the Data Protection Application with the `kubevirt` and `openshift`]
```
```suggestion
* xref:../../../backup_and_restore/application_backup_and_restore/installing/installing-oadp-kubevirt.adoc#oadp-installing-dpa-1-3_installing-oadp-kubevirt[Installing the Data Protection Application with the `kubevirt` and `openshift` plugins]
```

anarnold97 commented on backup_and_restore/application_backup_and_restore/installing/installing-oadp-aws.adoc:
```diff
@@ -35,4 +35,9 @@ include::modules/oadp-installing-dpa-1-2-and-earlier.adoc[leveloffset=+1]
 
+.Additional resources
+
+* xref:../../../backup_and_restore/application_backup_and_restore/installing/installing-oadp-kubevirt.adoc#oadp-installing-dpa-1-3_installing-oadp-kubevirt[Installing the Data Protection Application] with the `kubevirt` and `openshift` xref:../../../backup_and_restore/application_backup_and_restore/oadp-features-plugins.adoc#oadp-plugins_oadp-features-plugins[plugins].
```
```suggestion
* xref:../../../backup_and_restore/application_backup_and_restore/installing/installing-oadp-kubevirt.adoc#oadp-installing-dpa-1-3_installing-oadp-kubevirt[Installing the Data Protection Application with the `kubevirt` and `openshift` plugins] 
```

anarnold97 commented on backup_and_restore/application_backup_and_restore/installing/installing-oadp-azure.adoc:
```diff
@@ -35,4 +35,8 @@ include::modules/oadp-installing-dpa-1-2-and-earlier.adoc[leveloffset=+1]
 
+.Additional resources
+
+* xref:../../../backup_and_restore/application_backup_and_restore/installing/installing-oadp-kubevirt.adoc#oadp-installing-dpa-1-3_installing-oadp-kubevirt[Installing the Data Protection Application] with the `kubevirt` and `openshift` xref:../../../backup_and_restore/application_backup_and_restore/oadp-features-plugins.adoc#oadp-plugins_oadp-features-plugins[plugins].
```
```suggestion
* xref:../../../backup_and_restore/application_backup_and_restore/installing/installing-oadp-kubevirt.adoc#oadp-installing-dpa-1-3_installing-oadp-kubevirt[Installing the Data Protection Application with the `kubevirt` and `openshift` plugins] 
```

anarnold97 commented on backup_and_restore/application_backup_and_restore/installing/installing-oadp-gcp.adoc:
```diff
@@ -36,4 +36,9 @@ include::modules/oadp-gcp-wif-cloud-authentication.adoc[leveloffset=+1]
 
+.Additional resources
+
+* xref:../../../backup_and_restore/application_backup_and_restore/installing/installing-oadp-kubevirt.adoc#oadp-installing-dpa-1-3_installing-oadp-kubevirt[Installing the Data Protection Application] with the `kubevirt` and `openshift` xref:../../../backup_and_restore/application_backup_and_restore/oadp-features-plugins.adoc#oadp-plugins_oadp-features-plugins[plugins].
```
```suggestion
* xref:../../../backup_and_restore/application_backup_and_restore/installing/installing-oadp-kubevirt.adoc#oadp-installing-dpa-1-3_installing-oadp-kubevirt[Installing the Data Protection Application with the `kubevirt` and `openshift` plugins] 
```

anarnold97 commented on backup_and_restore/application_backup_and_restore/installing/installing-oadp-mcg.adoc:
```diff
@@ -41,4 +41,11 @@ include::modules/oadp-installing-dpa-1-2-and-earlier.adoc[leveloffset=+1]
+
+For more information on installing OADP to back up and restore virtual machines, see:
+
+* xref:../../../backup_and_restore/application_backup_and_restore/installing/installing-oadp-kubevirt.adoc#oadp-installing-dpa-1-3_installing-oadp-kubevirt[Installing the Data Protection Application] with the `kubevirt` and `openshift` xref:../../../backup_and_restore/application_backup_and_restore/oadp-features-plugins.adoc#oadp-plugins_oadp-features-plugins[plugins].
```
```suggestion
* xref:../../../backup_and_restore/application_backup_and_restore/installing/installing-oadp-kubevirt.adoc#oadp-installing-dpa-1-3_installing-oadp-kubevirt[Installing the Data Protection Application with the `kubevirt` and `openshift` plugins] 
```

anarnold97 commented on backup_and_restore/application_backup_and_restore/installing/installing-oadp-kubevirt.adoc:
```diff
@@ -0,0 +1,53 @@
+
+include::modules/install-and-configure-oadp-kubevirt.adoc[leveloffset=+1]
+
+
+// You can configure the Data Protection Application by setting Velero resource allocations or enabling self-signed CA certificates.
```
```suggestion
[role="_additional-resources"]
.Additional resources

. xref:../../../backup_and_restore/application_backup_and_restore/oadp-features-plugins.adoc#oadp-plugins_oadp-features-plugins[plugins]
. xref:../../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/backing-up-applications.adoc#backing-up-applications[`Backup` custom resource (CR)]
. xref:../../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/restoring-applications.adoc#restoring-applications[`Restore` CR]
.  xref:../../../operators/admin/olm-restricted-networks.adoc#olm-restricted-networks[Using Operator Lifecycle Manager on restricted networks]
```

anarnold97 commented on modules/install-and-configure-oadp-kubevirt.adoc:
```diff
@@ -0,0 +1,37 @@
+
+.Procedure
+
+. Install the xref:../../../backup_and_restore/application_backup_and_restore/installing/about-installing-oadp.adoc#about-installing-oadp[OADP Operator] according to the instructions for your storage provider.
```
```suggestion
. Install the {oadp-short} Operator according to the instructions for your storage provider.
```

anarnold97 commented on modules/install-and-configure-oadp-kubevirt.adoc:
```diff
@@ -0,0 +1,37 @@
+
+. Install the {oadp-short} Operator according to the instructions for your storage provider.
+
+. Install the xref:oadp-installing-dpa-1-3_installing-oadp-kubevirt[Data Protection Application (DPA)] with the `kubevirt` and `openshift` xref:../../../backup_and_restore/application_backup_and_restore/oadp-features-plugins.adoc#oadp-plugins_oadp-features-plugins[plugins].
```
```suggestion
. Install the Data Protection Application (DPA)] with the `kubevirt` and `openshift` {oadp-short} plugins.
```

anarnold97 commented on modules/install-and-configure-oadp-kubevirt.adoc:
```diff
@@ -0,0 +1,37 @@
+
+. Install the Data Protection Application (DPA)] with the `kubevirt` and `openshift` {oadp-short} plugins.
+
+. Back up virtual machines by creating a xref:../../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/backing-up-applications.adoc#backing-up-applications[`Backup` custom resource (CR)].
```
```suggestion
. Back up virtual machines by creating a `Backup` custom resource (CR).
```

anarnold97 commented on modules/install-and-configure-oadp-kubevirt.adoc:
```diff
@@ -0,0 +1,37 @@
+* CSI backups with DataMover.
+====
+
+You restore the `Backup` CR by creating a xref:../../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/restoring-applications.adoc#restoring-applications[`Restore` CR].
```
```suggestion
You restore the `Backup` CR by creating a `Restore` CR.
```

anarnold97 commented on modules/install-and-configure-oadp-kubevirt.adoc:
```diff
@@ -0,0 +1,37 @@
+
+You restore the `Backup` CR by creating a `Restore` CR.
+
+To install the {oadp-short} Operator in a restricted network environment, you must first disable the default OperatorHub sources and mirror the Operator catalog. For details, see xref:../../../operators/admin/olm-restricted-networks.adoc#olm-restricted-networks[Using Operator Lifecycle Manager on restricted networks].
```
```suggestion
To install the {oadp-short} Operator in a restricted network environment, you must first disable the default OperatorHub sources and mirror the Operator catalog. 
```

anarnold97 commented on backup_and_restore/application_backup_and_restore/installing/installing-oadp-mcg.adoc:
```diff
@@ -41,4 +41,11 @@ include::modules/oadp-installing-dpa-1-2-and-earlier.adoc[leveloffset=+1]
 include::modules/oadp-installing-dpa-1-3.adoc[leveloffset=+1]
 include::modules/oadp-enabling-csi-dpa.adoc[leveloffset=+2]
 
+[role="_additional-resources"]
+.Additional resources
+
+* link:https://access.redhat.com/solutions/6719951[Performance tuning guide for Multicloud Object Gateway].
+
+* xref:../../../backup_and_restore/application_backup_and_restore/installing/installing-oadp-kubevirt.adoc#oadp-installing-dpa-1-3_installing-oadp-kubevirt[Installing the Data Protection Application with the `kubevirt` and `openshift` plugins] 
```
```suggestion
```

anarnold97 commented on backup_and_restore/application_backup_and_restore/installing/installing-oadp-mcg.adoc:
```diff
@@ -41,4 +41,5 @@ include::modules/oadp-installing-dpa-1-2-and-earlier.adoc[leveloffset=+1]
 include::modules/oadp-installing-dpa-1-3.adoc[leveloffset=+1]
 include::modules/oadp-enabling-csi-dpa.adoc[leveloffset=+2]
 
+
```
```suggestion


[discrete]
[role="_additional-resources"]
.Additional resources

* link:https://access.redhat.com/solutions/6719951[Performance tuning guide for Multicloud Object Gateway].

* xref:../../../backup_and_restore/application_backup_and_restore/installing/installing-oadp-kubevirt.adoc#oadp-installing-dpa-1-3_installing-oadp-kubevirt[Installing the Data Protection Application with the `kubevirt` and `openshift` plugins] 

:installing-oadp-mcg!:
```

anarnold97 commented on backup_and_restore/application_backup_and_restore/installing/installing-oadp-mcg.adoc:
```diff
@@ -41,4 +41,5 @@ include::modules/oadp-installing-dpa-1-2-and-earlier.adoc[leveloffset=+1]
 include::modules/oadp-enabling-csi-dpa.adoc[leveloffset=+2]
 
+
 :installing-oadp-mcg!:
```
```suggestion
```

anarnold97 commented:
* Moved procedure to module
* Moved `xrefs` to `.Additional resources`
* Added level 2 heading... if you have a better name for this, then all ears

/label peer-review-needed

skrthomas commented on backup_and_restore/application_backup_and_restore/installing/installing-oadp-kubevirt.adoc:
```diff
@@ -0,0 +1,58 @@
+[role="_additional-resources"]
+.Additional resources
+
+. xref:../../../backup_and_restore/application_backup_and_restore/oadp-features-plugins.adoc#oadp-plugins_oadp-features-plugins[plugins]
+. xref:../../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/backing-up-applications.adoc#backing-up-applications[`Backup` custom resource (CR)]
+. xref:../../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/restoring-applications.adoc#restoring-applications[`Restore` CR]
+.  xref:../../../operators/admin/olm-restricted-networks.adoc#olm-restricted-networks[Using Operator Lifecycle Manager on restricted networks]
```
```suggestion
* xref:../../../backup_and_restore/application_backup_and_restore/oadp-features-plugins.adoc#oadp-plugins_oadp-features-plugins[plugins]
* xref:../../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/backing-up-applications.adoc#backing-up-applications[`Backup` custom resource (CR)]
* xref:../../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/restoring-applications.adoc#restoring-applications[`Restore` CR]
* xref:../../../operators/admin/olm-restricted-networks.adoc#olm-restricted-networks[Using Operator Lifecycle Manager on restricted networks]
```

skrthomas replied:
These should be in an unordered list per the .Additional resources guidelines

skrthomas commented on modules/install-and-configure-oadp-kubevirt.adoc:
```diff
@@ -0,0 +1,37 @@
+// backup_and_restore/application_backup_and_restore/installing/installing-oadp-kubevirt.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="install-and-configure-oadp-kubevirt"]
```
```suggestion
[id="install-and-configure-oadp-kubevirt_{context}"]
```

skrthomas commented on modules/install-and-configure-oadp-kubevirt.adoc:
```diff
@@ -0,0 +1,37 @@
+// Module included in the following assemblies:
+//
+// backup_and_restore/application_backup_and_restore/installing/installing-oadp-kubevirt.adoc
```
```suggestion
// * backup_and_restore/application_backup_and_restore/installing/installing-oadp-kubevirt.adoc
```

skrthomas reviewed (commented):
Just a few things to tidy up here! Looking good. Thanks for addressing those comments. I feel that this should be OK after this round.

skrthomas commented on modules/oadp-installing-dpa-1-3.adoc:
```diff
@@ -321,7 +321,7 @@ spec:
 endif::[]
 
-ifdef::virt-installing-configuring-oadp[]
+ifdef::installing-oadp-kubevirt,virt-installing-configuring-oadp[]
```
Actually sorry for the post-review comment but I don't see an endif:: for this. Am I missing something? It kinda missed my review due to the lines being collapsed but I don't see it after expanding.

skrthomas commented on modules/oadp-installing-dpa-1-3.adoc:
```diff
@@ -321,7 +321,7 @@ spec:
 endif::[]
 
-ifdef::virt-installing-configuring-oadp[]
+ifdef::installing-oadp-kubevirt,virt-installing-configuring-oadp[]
```
Can you provide the other preview link to compare for this ifdef?

anarnold97 replied:
https://72372--ocpdocs-pr.netlify.app/openshift-enterprise/latest/virt/backup_restore/virt-installing-configuring-oadp#oadp-installing-dpa-1-3_virt-installing-configuring-oadp

skrthomas replied:
Thank you!

maxwelldb commented on backup_and_restore/application_backup_and_restore/installing/installing-oadp-kubevirt.adoc:
```diff
@@ -0,0 +1,58 @@
+
+include::modules/oadp-installing-dpa-1-3.adoc[leveloffset=+1]
+
+[IMPORTANT]
```
Would it be more helpful for this important admonition to be toward the top of the page?

anarnold97 commented on backup_and_restore/application_backup_and_restore/installing/installing-oadp-kubevirt.adoc:
```diff
@@ -0,0 +1,58 @@
+[role="_additional-resources"]
+.Additional resources
+
+. xref:../../../backup_and_restore/application_backup_and_restore/oadp-features-plugins.adoc#oadp-plugins_oadp-features-plugins[OADP plugins]
```
```suggestion
. xref:../../../backup_and_restore/application_backup_and_restore/oadp-features-plugins.adoc#oadp-plugins_oadp-features-plugins[{oadp-short} plugins]
```

anarnold97 commented on backup_and_restore/application_backup_and_restore/installing/installing-oadp-ocs.adoc:
```diff
@@ -46,5 +46,12 @@ include::modules/oadp-installing-dpa-1-3.adoc[leveloffset=+1]
 
+.Additional resources
+
+For more information on installing OADP to back up and restore virtual machines, see:
```
```suggestion
```

anarnold97 commented:
/label merge-review-needed

maxwelldb commented on backup_and_restore/application_backup_and_restore/installing/installing-oadp-kubevirt.adoc:
```diff
@@ -0,0 +1,58 @@
+
+The following storage options are excluded:
+
+* File System backup and restore.
+
+* Volume Snapshot Backups and Restores.
```
```suggestion
* File system backup and restore

* Volume snapshot backups and restores
```

- Remove periods unless introductory phrase + list items form complete sentences. 

- Sentence case.

maxwelldb commented on backup_and_restore/application_backup_and_restore/installing/installing-oadp-kubevirt.adoc:
```diff
@@ -0,0 +1,58 @@
+// If docs team approves the name, "Configuring OADP with Kubevirt" great but I suspect it will need to be "Configuring OADP with OpenShift Virtualization" or "Configuring OADP with Virtualization" or perhaps... bear with me.. "OADP and OpenShift-Virtualization"
+= Configuring the {oadp-full} with {VirtProductName}
+include::_attributes/common-attributes.adoc[]
+:context: installing-oadp-kubevirt
+:installing-oadp-kubevirt:
+:credentials: cloud-credentials
+:provider: kubevirt
```
What is the intent of these attributes?

maxwelldb commented on backup_and_restore/application_backup_and_restore/installing/installing-oadp-kubevirt.adoc:
```diff
@@ -0,0 +1,58 @@
+
+For more information, see xref:../../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/oadp-backing-up-applications-restic-doc.adoc#oadp-backing-up-applications-restic-doc[Backing up applications with File System Backup: Kopia or Restic].
+====
+
```
```suggestion
```

maxwelldb commented on backup_and_restore/application_backup_and_restore/installing/installing-oadp-ocs.adoc:
```diff
@@ -46,5 +46,11 @@ include::modules/oadp-installing-dpa-1-3.adoc[leveloffset=+1]
 
+.Additional resources
+
+
```
```suggestion
```

maxwelldb commented on backup_and_restore/application_backup_and_restore/installing/installing-oadp-kubevirt.adoc:
```diff
@@ -0,0 +1,58 @@
+[role="_additional-resources"]
+.Additional resources
+
+. xref:../../../backup_and_restore/application_backup_and_restore/oadp-features-plugins.adoc#oadp-plugins_oadp-features-plugins[{oadp-short} plugins]
+. xref:../../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/backing-up-applications.adoc#backing-up-applications[`Backup` custom resource (CR)]
+. xref:../../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/restoring-applications.adoc#restoring-applications[`Restore` CR]
+. xref:../../../operators/admin/olm-restricted-networks.adoc#olm-restricted-networks[Using Operator Lifecycle Manager on restricted networks]
```
This must be an unordered list.

```suggestion
* xref:../../../backup_and_restore/application_backup_and_restore/oadp-features-plugins.adoc#oadp-plugins_oadp-features-plugins[{oadp-short} plugins]
* xref:../../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/backing-up-applications.adoc#backing-up-applications[`Backup` custom resource (CR)]
* xref:../../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/restoring-applications.adoc#restoring-applications[`Restore` CR]
* xref:../../../operators/admin/olm-restricted-networks.adoc#olm-restricted-networks[Using Operator Lifecycle Manager on restricted networks]
```

maxwelldb commented on modules/install-and-configure-oadp-kubevirt.adoc:
```diff
@@ -0,0 +1,37 @@
+
+.Procedure
+
+. Install the {oadp-short} Operator according to the instructions for your storage provider.
```
This procedure doesn't make sense to me. 

Description: "As a cluster administrator, you install {oadp-short} by installing the {oadp-short} Operator."

Step 1: "Install the {oadp-short} Operator according to the instructions for your storage provider."

It doesn't actually tell me how to do what the title indicates.

anarnold97 replied:
It has been approved multiple times

maxwelldb commented on modules/install-and-configure-oadp-kubevirt.adoc:
```diff
@@ -0,0 +1,37 @@
+
+You restore the `Backup` CR by creating a `Restore` CR.
+
+To install the {oadp-short} Operator in a restricted network environment, you must first disable the default OperatorHub sources and mirror the Operator catalog. 
```
This should be at the top of the procedure somewhere so that readers who are working in restricted environments don't go down the wrong path.

anarnold97 replied:
It has been approved multiple times

maxwelldb replied:
I'm a layer of that approval. Would this be good for someone in this environment to know up front or not? @anarnold97

skrthomas replied:
I'm sorry @anarnold97, peer review is really a collective effort. None of us are full time editors, and this is a good detail that I missed. To back up the importance, its well-known that restricted environment users already struggle a lot.

maxwelldb commented on backup_and_restore/application_backup_and_restore/installing/installing-oadp-kubevirt.adoc:
```diff
@@ -0,0 +1,58 @@
+
+include::modules/oadp-installing-dpa-1-3.adoc[leveloffset=+1]
+
+[IMPORTANT]
```
If this is important, why is it the last thing in the assembly? If it is something that a reader needs to know about this process, hoist it up.

skrthomas replied:
+1 Max. A support statement is important and lobbing it at the bottom could lead customers down a path they can't actually take in the end.

maxwelldb reviewed (commented):
Left comments, suggestions, and questions from merge review. Relabel when ready.

skrthomas commented on backup_and_restore/application_backup_and_restore/installing/installing-oadp-gcp.adoc:
```diff
@@ -36,4 +36,9 @@ include::modules/oadp-gcp-wif-cloud-authentication.adoc[leveloffset=+1]
 include::modules/oadp-installing-dpa-1-3.adoc[leveloffset=+1]
 include::modules/oadp-enabling-csi-dpa.adoc[leveloffset=+2]
 
+.Additional resources
```
```suggestion
[role="_additional-resources"]
.Additional resources
```

maxwelldb commented on backup_and_restore/application_backup_and_restore/installing/installing-oadp-ocs.adoc:
```diff
@@ -46,5 +46,11 @@ include::modules/oadp-installing-dpa-1-3.adoc[leveloffset=+1]
 include::modules/oadp-creating-object-bucket-claim.adoc[leveloffset=+2]
 include::modules/oadp-enabling-csi-dpa.adoc[leveloffset=+2]
 
+.Additional resources
```
```suggestion
[role="_additional-resources"]
.Additional resources
```

anarnold97 replied:
It has been approved multiple times

anarnold97 commented:
/label merge-review-needed

anarnold97 commented on modules/install-and-configure-oadp-kubevirt.adoc:
```diff
@@ -0,0 +1,37 @@
+
+You restore the `Backup` CR by creating a `Restore` CR.
+
+To install the {oadp-short} Operator in a restricted network environment, you must first disable the default OperatorHub sources and mirror the Operator catalog. 
```
```suggestion
```

anarnold97 commented:
Updated

skrthomas commented on backup_and_restore/application_backup_and_restore/installing/installing-oadp-ocs.adoc:
```diff
@@ -46,5 +46,11 @@ include::modules/oadp-installing-dpa-1-3.adoc[leveloffset=+1]
+[role="_additional-resources"]
+.Additional resources
+
+* xref:../../../backup_and_restore/application_backup_and_restore/installing/installing-oadp-kubevirt.adoc#oadp-installing-dpa-1-3_installing-oadp-kubevirt[Installing the Data Protection Application] with the `kubevirt` and `openshift` xref:../../../backup_and_restore/application_backup_and_restore/oadp-features-plugins.adoc#oadp-plugins_oadp-features-plugins[plugins].
```
I unresolved a previously resolved global comment about this, but the second xref needs to be its own *, and there shouldn't be any text outside of the xref []
```suggestion
* xref:../../../backup_and_restore/application_backup_and_restore/installing/installing-oadp-kubevirt.adoc#oadp-installing-dpa-1-3_installing-oadp-kubevirt[Installing the Data Protection Application]
*  xref:../../../backup_and_restore/application_backup_and_restore/oadp-features-plugins.adoc#oadp-plugins_oadp-features-plugins[plugins].
```

anarnold97 commented:
@weshayutin - i am sorry i am going to close this PR as I need to speak to my line manager. 

The irony is that this documentation was merely copying `openshift-docs/virt/backup_restore/virt-backup-restore-overview.adoc` as part of moving Virt docs for backup and restore into OADP`.
